docs/content/014-guides/022-source-types.mdx

```diff
@@ -0,0 +1,449 @@
+                                       // |
+queryType({                            // |
+  definition(t) {                      // |
+    t.list.field('users', {            // |
+      type: 'User',                    // |
+      resolve() {                      // |
+        return [/**/]                  // |
+//               ^----------------------- |
+      },
+    })
+  },
+})
+```
+
+This may suffice well enough for some time, but most apps will eventually see their GraphQL and source types diverge. Once this happens, you can tell Nexus about it using one of the solutions explained below.
+
+### Globally configure source types
+
+Configuring your source types globally is the recommended way in most cases. Typically, your source types will come from your database model types (also called entities), or will be generated by your ORM.
+
+When that's the case, you can configure Nexus to automatically map your source types to your GraphQL Object types based on their name.
+
+This is all done in the `typegenAutoConfig` option of `makeSchema`.
+
+#### Exact name matching
+
+By default, Nexus will automatically map the source type `User` to a GraphQL Object type also named `User`. Let's see how that works:
+
+```ts
+// ./db.ts
+
+export interface User {
+// ^-------------^---------- Export your source type (required)
+//               |---------- Name of your source type which will be extracted by Nexus
+  firstName: string
+  lastName: string
+}
+
+export interface Post {
+// ^-------------^---------- Export your source type (required)
+//               |---------- Name of your source type which will be extracted by Nexus
+// ...
+}
+
+export interface Comment {
+// ^-------------^---------- Export your source type (required)
+//               |---------- Name of your source type which will be extracted by Nexus
+// ...
+}
+
+// ./schema.ts
+import { makeSchema, objectType, queryType } from '@nexus/schema'
+import * as path from 'path'
+
+const User = objectType({
+  name: 'User',
+  //      ^---------------------- Name of your GraphQL Object type which will be matched against the name of your source types
+  definition(t) {
+    t.string('fullName', {
+      resolve(user) {
+        //      ^------------------------ as: User
+        return user.firstName + ' ' + user.lastName
+      }
+    })
+  }
+})
+
+const Query = queryType({
+  definition(t) {
+    t.list.field('users', {
+      type: User,
+      resolve(_parent, args, ctx) {
+        //                       ^------- return as: User[]
+        // ...
+      }
+    })
+  }
+})
+
+const schema = makeSchema({
+  types: [User, Query],
+  typegenAutoConfig: {
+    sources: [{
+      source: path.join(__dirname, 'db.ts'),
+      //       ^----------------------------- Path to the file containing your source types
+      alias: 'db'
+      //      ^------------------------------ Arbitrary unique name used as alias for Nexus to import your source types. eg: import * as <alias> from <source>
```

Comment:
Let's try to keep within bounds of codeblock:

![image](https://user-images.githubusercontent.com/284476/101397096-404f7400-389a-11eb-881b-0bbb8a95544f.png)